enacts/crop_suitability/maproom_crop_suit.py

```diff
@@ -0,0 +1,553 @@
+    State("temp_range","value"),
+    State("season_length","value"),
+    State("min_wet_days","value"),
+    State("wet_day_def","value"),
+    State("number_dry_spells","value"),
+    State("dry_days_in_row","value"),
+    State("dry_spell_rain","value"),
+)
+def timeseries_plot(
+    loc_marker,
+    data_choice,
+    n_clicks,
+    target_year,
+    target_season,
+    lower_wet_threshold,
+    upper_wet_threshold,
+    minimum_temp,
+    maximum_temp,
+    temp_range,
+    season_length,
+    min_wet_days,
+    wet_day_def,
+    number_dry_spells,
+    dry_days_in_row,
+    dry_spell_rain
+):
+    lat1 = loc_marker[0]
+    lng1 = loc_marker[1]
+
+    if data_choice == "suitability_map":
+        data = crop_suitability(
+            rr_mrg, min_wet_days, wet_day_def, tmax_mrg, tmin_mrg, 
+            lower_wet_threshold, upper_wet_threshold, maximum_temp,
+            minimum_temp, temp_range, target_season, dry_spell_rain,
+            dry_days_in_row,number_dry_spells)
+    if data_choice == "precip_map":
+        data = rr_mrg
+    if data_choice == "tmax_map":
+        data = tmax_mrg
+    if data_choice == "tmin_map":
+        data = tmin_mrg
+    
+    try:
+        if data_choice == "precip_map":
+            data_var = pingrid.sel_snap(data.precip, lat1, lng1)
+            isnan = np.isnan(data_var).sum().sum()
+        elif data_choice == "suitability_map":
+            data_var = pingrid.sel_snap(data.crop_suit, lat1, lng1)
+            isnan = np.isnan(data_var).sum().sum()            
+        else:
+            data_var = pingrid.sel_snap(data.temp, lat1, lng1)
+            isnan = np.isnan(data_var).sum().sum()
+        if isnan > 0:
+            error_fig = pingrid.error_fig(error_msg="Data missing at this location")
+            germ_sentence = ""
+            return error_fig, error_fig, germ_sentence
+    except KeyError:
+        error_fig = pingrid.error_fig(error_msg="Grid box out of data domain")
+        germ_sentence = ""
+        return error_fig, error_fig, germ_sentence
+
+    if data_choice == "suitability_map":
+        seasonal_suit = data_var
+        timeseries_plot = pgo.Figure()
+        timeseries_plot.add_trace(
+            pgo.Scatter(
+                x = seasonal_suit["year"].values,
+                y = seasonal_suit.values,
+                line=pgo.scatter.Line(color="blue"),
+            )
+        )
+        timeseries_plot.update_traces(mode="lines", connectgaps=False)
+        timeseries_plot.update_layout(
+            xaxis_title = "years",
+            yaxis_title = f"{CONFIG['map_text'][data_choice]['data_var']} ({CONFIG['map_text'][data_choice]['units']})",
+            title = f"{CONFIG['map_text'][data_choice]['menu_label']} seasonal climatology timeseries plot"
+        ) 
+    else:
+        data_var.load()
+    
+        seasonal_var = data_var.sel(T=data_var['T.season']==target_season)
+        seasonal_mean = seasonal_var.groupby("T.year").mean("T")
```

Comment:
it looks like this is the only difference with the case crop_suitability, since you are checking on the case earlier, maybe you can do that there and thus you need not have if cases to make the graph?